ext/mysqlnd/mysqlnd_auth.c

```diff
@@ -99,10 +99,7 @@ mysqlnd_run_authentication(
 			switch_to_auth_protocol = NULL;
 			switch_to_auth_protocol_len = 0;
 
-			if (conn->authentication_plugin_data.s) {
-				mnd_pefree(conn->authentication_plugin_data.s, conn->persistent);
-				conn->authentication_plugin_data.s = NULL;
-			}
+			mysqlnd_set_persistent_string(&conn->authentication_plugin_data, NULL, 0, conn->persistent);
 			conn->authentication_plugin_data.l = plugin_data_len;
 			conn->authentication_plugin_data.s = mnd_pemalloc(conn->authentication_plugin_data.l, conn->persistent);
 			memcpy(conn->authentication_plugin_data.s, plugin_data, plugin_data_len);
```

Comment:
I am not sure If I can change this

Reply:
pemalloc+memcpy is the same as pestrndup, so changing this should be fine.

Reply:
Ok, I tried it but I get warnings because it is `zend_uchar`. I am not sure what is going on with this assignment so I will roll back this one change.